File: test_project/gen_code_cs/tbitem.cs
namespace demo
{

    public class TbItem
    {
        private readonly System.Collections.Generic.Dictionary<int, Item> _dataMap;
        private readonly System.Collections.Generic.List<Item> _dataList;
        
        public TbItem(System.Collections.Generic.List<Item> dataList)
        {
            _dataList = dataList;
            _dataMap = new System.Collections.Generic.Dictionary<int, Item>();
            foreach(var v in _dataList)
            {
                _dataMap.Add(v.Id, v);
            }
        }
        
        public System.Collections.Generic.Dictionary<int, Item> DataMap => _dataMap;
        public System.Collections.Generic.List<Item> DataList => _dataList;
        
        public Item GetOrDefault(int key)
        {
            if (_dataMap.TryGetValue(key, out var v))
            {
                return v;
            }
            return null;
        }
        
        public Item Get(int key)
        {
            if (_dataMap.TryGetValue(key, out var v))
            {
                return v;
            }
            throw new System.Collections.Generic.KeyNotFoundException();
        }
    }

}

File: test_project/gen_code_ts/tbitem.ts
export namespace demo {

    export class TbItem {
    private readonly _dataMap: Map<number, Item>;
    private readonly _dataList: Item[];
    
    constructor(dataList: Item[]) {
        this._dataList = dataList;
        this._dataMap = new Map<number, Item>();
        for (const v of this._dataList) {
            this._dataMap.set(v.Id, v);
        }
    }
    
    public get dataMap(): Map<number, Item> {
        return this._dataMap;
    }
    
    public get dataList(): Item[] {
        return this._dataList;
    }
    
    public get(key: number): Item | undefined {
        return this._dataMap.get(key);
    }
}

}
